test/date_time_parser_test.rb

```diff
@@ -53,4 +53,13 @@
     @date_parser.parse(date).must_equal date
     @time_parser.parse(time).must_equal time
   end
+
+  it "raises ArgumentError when parsing a bad string" do
+    proc { @date_parser.parse("asdf") }.must_raise ArgumentError
```

Comment:
Bah, that's fucked up. But at least it gives us leeway: Since Ruby itself doesn't seem to be consistent, I have no problem with using our own error message here. And I think a good error message should be as helpful as possible – so in our case, I think we definitely think we should include the supplied date, right?